tools/clang/lib/AST/ASTContext.cpp

```diff
@@ -8461,7 +8461,10 @@ bool ASTContext::DeclMustBeEmitted(const Decl *D) {
     if (Linkage == GVA_Internal || Linkage == GVA_AvailableExternally ||
         Linkage == GVA_DiscardableODR)
       return false;
-    return true;
+    // HLSL Change Starts
+    // Don't just return true because of visibility, unless building a library (which is not currently implemented)
+    return FD->getName() == getLangOpts().HLSLEntryFunction || IsPatchConstantFunctionDecl(FD);
```

Comment:
Not right now. ASTContext doesn't have visibility into the target profile. We'll have to add it when we add library support (and then at that point we'll be revisiting this bit of code in any case).